pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -23,8 +23,8 @@ option java_package = "org.apache.pulsar.common.api.proto";
 option optimize_for = LITE_RUNTIME;
 
 message MessageIdData {
-	required uint64 ledgerId = 1;
-	required uint64 entryId  = 2;
+	required int64 ledgerId = 1;
+	required int64 entryId  = 2;
```

Comment:
This change is safe as it doesn't change the encoding on the wire, just the type used in C++. In Java it would anyway be a `long` with no difference. 
This is required to pass `-1` as special values.